ratatui-core/src/layout/rect.rs

```diff
@@ -351,6 +353,35 @@ impl Rect {
         }
     }
 
+    /// Centers the rect horizontally. The constraint indicates the result rects width.
```

Comment:
The usual rust (and often other languages) convention for doc comments is simple single sentence as the first line. This appears anywhere where a short description is needed. The detail generally is separated by a newline and then more information that talks about how this works.

```suggestion
    /// Returns a new Rect, centered horizontally based on the provided constraint.
```

It would be good to add a small example of use here. E.g. something like:

```rust
fn render(frame &mut Frame)
    let area = frame.area().centered_horizontally(Constraint::Ratio(1, 2));
}
```

(This comment also applies to the other methods below)